### DISCUSSION TITLE
DNS resolution problem

### DISCUSSION THREAD

**Discussion by Joci78** [arendst/Tasmota#19393] [General]
Dear community,

I have a test environment:
Network: 192.168.10.0/24
Mosquitto server: 192.168.10.120
Router: 192.168.10.254 (pFsense)

The sensors receive an IP via DHCP. The system works without problems. I started setting up TLS communication. I set the DNS name of the server in the DNS resolver on the router (192.168.10.120 -> mqtt.home.local). Name resolution works fine from a workstation in the network. Unfortunately, it does not work in ESP. The DNS server (pFsense) and the network are the same as on the workstation.

Comment: Comment: I set cert to domain name, not IP.

Has anyone experienced this? What can be the solution for it? I don't want to use IP in common name for cert.

**Comment by s-hadinger** [comment]
Domain in `.local` are mDNS and not real dns. Tasmota does not support mDNS by default because it creates too many problems. 

I also doubt that mDNS domain names are valid TLS domains. For TLS you need a real dns domain name.

**Reply by Joci78** [reply]
Thanks for the reply. I need to rethink the system.
